src/Microsoft.OData.Client/Attribute/UriFunctionAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="UriFunctionAttribute.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Client
+{
+    using System;
+
+    /// <summary>Indicates a method that should be mapped to a Uri Function.</summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public sealed class UriFunctionAttribute : Attribute
+    {
+        /// <summary>Allow client side evaluation.</summary>
+        private readonly bool canBeEvaluated;
+
+        /// <summary>Initializes a new instance of the <see cref="Microsoft.OData.Client.UriFunctionAttribute" /> class. </summary>
+        /// <param name="canBeEvaluated">Use client side evaluation when possible. Default is false.</param>
+        public UriFunctionAttribute(bool canBeEvaluated = false)

```

Comment:
For a method (lets say a static one) having `canBeEvaluated=true` it is the parameters that will decide if it is evaluated on the client or send to the server. If parameters are constants then it can be evaluated on the client and will be when `canBeEvaluated=true`.

The build in UriFunctions behaves the same way - they are evaluated on client if parameters are constant. 
So basically it is the opposite situation that is interesting - getting the UriFunction to be forced to evaluate on the server. An example of where this can be relevant is the standard V4 UriFuction `now()` - I will really like that to be resolved on the server and by defining it in the client and leaving out `canBeEvaluated=true` I can now finally have server `now()`.

I will change the name to allowClientSideEvaluation.
Thanks.

Reply:
Thanks for the clarification.

Reply:
Pushed change. Thanks.